src/pages/company-info/index.js

```diff
@@ -0,0 +1,232 @@
+import { Box, Grid } from '@mui/material'
+import React, { useContext, useEffect, useState } from 'react'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import FormShell from 'src/components/Shared/FormShell'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import useResourceParams from 'src/hooks/useResourceParams'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import toast from 'react-hot-toast'
+import { useForm } from 'src/hooks/form'
+import ImageUpload from 'src/components/Inputs/ImageUpload'
+import { getStorageData } from 'src/storage/storage'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+
+export function PostImage(postRequest, obj, initialValues) {
```

Comment:
Move this into ImageUpload. Use forwardRef and useImperativeHandle to pass the submit function back to the parent.